Mathlib/Algebra/MonoidAlgebra/Basic.lean

```diff
@@ -1434,61 +1483,99 @@ instance nontrivial [Semiring k] [Nontrivial k] [Nonempty G] : Nontrivial k[G] :
 section DerivedInstances
 
 instance commSemiring [CommSemiring k] [AddCommMonoid G] : CommSemiring k[G] :=
-  { AddMonoidAlgebra.nonUnitalCommSemiring, AddMonoidAlgebra.semiring with }
+  { mul_comm := mul_comm }
 #align add_monoid_algebra.comm_semiring AddMonoidAlgebra.commSemiring
 
 instance unique [Semiring k] [Subsingleton k] : Unique k[G] :=
   Finsupp.uniqueOfRight
 #align add_monoid_algebra.unique AddMonoidAlgebra.unique
 
+instance neg [Ring k] : Neg k[G] :=
+  Finsupp.neg
+
+instance sub [Ring k] : Sub k[G] :=
+  Finsupp.sub
+
 instance addCommGroup [Ring k] : AddCommGroup k[G] :=
-  Finsupp.addCommGroup
+  --TODO: add reference to library note in PR #7432
+  { Finsupp.addCommGroup with
```

Comment:
Similar here. This seems like it should be a problem with `Finsupp.addCommGroup` primarily.